src/app/shared/components/catalog-filter/catalog-filter.component.ts

```diff
@@ -0,0 +1,27 @@
+import { ChangeDetectionStrategy, Component, HostListener, OnInit } from '@angular/core';
+import { CatalogFilterService } from '@app/catalog/services/catalog/catalog-filter.service';
+import { UntilDestroy, untilDestroyed } from '@shared/until-destroy/until-destroy';
+
+@UntilDestroy
+@Component({
+  selector: 'lc-catalog-filter',
+  templateUrl: './catalog-filter.component.html',
+  styleUrls: ['./catalog-filter.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class CatalogFilterComponent implements OnInit {
+  @HostListener('click') onClick(): void {
+    this.catalogFilterService.setFiltersButtonToggled(true);
```

Comment:
I think we can open modal after click on it derectly.
I think we do not need set toggle subject at all.